CodeSniffer/Standards/Generic/Sniffs/PHP/SyntaxSniff.php

```diff
@@ -74,11 +74,11 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
         }
 
         $fileName = $phpcsFile->getFilename();
-        $cmd      = $this->_phpPath." -l \"$fileName\" 2>&1";
+        $cmd      = $this->_phpPath." -l -d error_prepend_string='' \"$fileName\" 2>&1";
```

Comment:
In that case I would rather change regex to figure out where errors are in the text instead of matching from text start. That would be more error prone.

For example the `on line ([0-9]+)` part of regex uniquely identifies the error. This way we don't care where we find it and know for 100% that error text comes before it.